### PR TITLE
feat(platform): controller support crd mode

### PR DESCRIPTION
**Pull request by leoryu**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/platform/controller/cluster/cluster_controller.go [hunk 1]
```diff
@@ -487,10 +494,20 @@ func (c *Controller) ensureCreateClusterCredential(ctx context.Context, cluster
 	}
 
 	// TODO use informer search by labels.
+	var clustercredentials *platformv1.ClusterCredentialList
 	fieldSelector := fields.OneTermEqualSelector("clusterName", cluster.Name).String()
 	clustercredentials, err := c.platformClient.ClusterCredentials().List(ctx, metav1.ListOptions{FieldSelector: fieldSelector})
 	if err != nil {
-		return nil, err
+		// if fieldSelector not support, try to use label selector
```
Comment (huxiaoliang) [inline pkg/platform/controller/cluster/cluster_controller.go:501]: no need call twice, just check cluster type is ok

File: pkg/platform/controller/cluster/config/types.go [hunk 2]
```diff
@@ -39,4 +39,6 @@ type ClusterControllerConfiguration struct {
 	BucketRateLimiterLimit int
 	// BucketRateLimiterBurst bursts of at most b tokens.
 	BucketRateLimiterBurst int
+	// IsCRDMode Whether the controller is using CRD mode")
```
Comment (huxiaoliang) [inline pkg/platform/controller/cluster/config/types.go:42]: remove ")

File: pkg/platform/controller/cluster/cluster_controller.go [hunk 3]
```diff
@@ -345,6 +348,9 @@ func (c *Controller) reconcile(ctx context.Context, key string, cluster *platfor
 	var err error
 
 	switch cluster.Status.Phase {
+	case "":
```
Comment (huxiaoliang) [inline pkg/platform/controller/cluster/cluster_controller.go:351]: add comments here for cluster.Status.Phase is empt string case

File: pkg/platform/controller/cluster/cluster_controller.go [hunk 4]
```diff
@@ -389,15 +395,28 @@ func (c *Controller) onCreate(ctx context.Context, cluster *platformv1.Cluster)
 		if err != nil {
 			// Update status, ignore failure
 			_, _ = c.platformClient.ClusterCredentials().Update(ctx, clusterWrapper.ClusterCredential, metav1.UpdateOptions{})
-			_, _ = c.platformClient.Clusters().Update(ctx, clusterWrapper.Cluster, metav1.UpdateOptions{})
+			updatedCls, _ := c.platformClient.Clusters().Update(ctx, clusterWrapper.Cluster, metav1.UpdateOptions{})
```
Comment (huxiaoliang) [inline pkg/platform/controller/cluster/cluster_controller.go:398]: add comments to used isCRDMode to distinguish update logic for crd add aa mode

**Review by huxiaoliang — COMMENTED** [review]
need new pr

File: cmd/tke-platform-controller/app/platform.go [hunk 5]
```diff
@@ -53,6 +53,7 @@ func startClusterController(ctx ControllerContext) (http.Handler, bool, error) {
 		ctx.InformerFactory.Platform().V1().Clusters(),
 		ctx.Config.ClusterController,
 		platformv1.ClusterFinalize,
+		ctx.Config.ClusterController.IsCRDMode,
```
Comment (wl-chen) [inline cmd/tke-platform-controller/app/platform.go:56]: 这个值可以从 ctx.Config.ClusterController 里拿吧，不需要再单独传进去

**Pull request merged**